### PR TITLE
Aiotemp dev

### PR DESCRIPTION
**Pull request by August-Ghost**
Add tempfile API support, now you can access the tempfile like this:
```python
async with AioTemporaryFile("w+") as f:
    await f.write("+++")
    await f.seek(0)
    print(await f.read())
```
And I moved wrap() to a separated file in order to wrap an opened file.

### CODE REVIEW

File: aiofiles/__init__.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 """Utilities for asyncio-friendly file handling."""
 from .threadpool import open
-
+from .aiotempfile import NamedAioTemporaryFile, AioTemporaryFile
+from .wrap import wrap
 __version__ = '0.3.2'
 
-__all__ = (open, )
+__all__ = (open, NamedAioTemporaryFile, AioTemporaryFile, wrap)
```
Comment (asvetlov) [inline aiofiles/__init__.py]: Adding `wrap` to public API maybe makes sense (the final decision is up to @Tinche sure) but if the function is a part of public API -- it should be documented. Right now aiofiles has no sphinx documentation (but I love to see aiofiles docs on readthedocs.io) -- but at least README should be updated.

File: aiofiles/__init__.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 """Utilities for asyncio-friendly file handling."""
 from .threadpool import open
-
+from .aiotempfile import NamedAioTemporaryFile, AioTemporaryFile
```
Comment (asvetlov) [inline aiofiles/__init__.py]: `aio` prefix is too redundant IMHO.
Just `aiofiles.tempfile` and  `NamedTemporaryFile`/`TemporaryFile` would be enough.

File: aiofiles/aiotempfile.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+from .base import AiofilesContextManager
+from functools import partial
+from tempfile import NamedTemporaryFile, TemporaryFile
+from .threadpool.binary import AsyncFileIO
+import asyncio
+
+
+def AioTemporaryFile(mode='w+b', buffering=-1, encoding=None,
+                      newline=None, suffix=None, prefix=None,
+                      dir=None, ):
```
Comment (asvetlov) [inline aiofiles/aiotempfile.py]: Inaccurate formatting

File: aiofiles/aiotempfile.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+def NamedAioTemporaryFile(mode='w+b', buffering=-1, encoding=None,
+                      newline=None, suffix=None, prefix=None,
+                      dir=None, delete=True):
+    return AiofilesContextManager(_wrap(NamedTemporaryFile, loop=None, executor=None, mode=mode,
+                 buffering=buffering, encoding=encoding,
+                      newline=newline, suffix=suffix, prefix=prefix,
+                      dir=dir,  delete=delete))
+
+
+def _wrap(callable, loop=None, executor=None, *args, **kargs):
```
Comment (asvetlov) [inline aiofiles/aiotempfile.py]: Why do you accept `loop` parameter but never use it in `_wrap` calls?

**Comment by Tinche** [comment]
We got this from another PR, thanks though!

**Pull request closed without merging**